drivers/vdpa/virtio/virtio_vdpa.c

```diff
@@ -0,0 +1,955 @@
+						   priv->vrings[index])) {
+			rte_intr_fd_set(intr_handle, -1);
+			DRV_LOG(ERR, "Failed to register virtq %d interrupt.",
+				index);
+			goto error;
+		} else {
+			DRV_LOG(DEBUG, "Register fd %d interrupt for virtq %d.",
+				rte_intr_fd_get(intr_handle),
+				index);
+		}
+	}
+
+	return 0;
+error:
+	virtio_vdpa_virtq_doorbell_relay_disable(priv, index);
+	return -EINVAL;
+}
+
+static int
+virtio_vdpa_virtq_disable(struct virtio_vdpa_priv *priv, int index)
+{
+	int ret;
+
+	ret = virtio_vdpa_virtq_doorbell_relay_disable(priv, index);
+	if (ret) {
+		 DRV_LOG(ERR, "Virtq doorbell relay failed");
+		 return ret;
+	}
+
+	virtio_pci_dev_queue_del(priv->vpdev, index);
+
+	priv->vrings[index]->notifier_state = VIRTIO_VDPA_NOTIFIER_STATE_DISABLED;
+
+	ret = virtio_pci_dev_interrupts_disable(priv->vpdev, index+1);
+	if (ret) {
+		 DRV_LOG(ERR, "Virtq %d interrupt disabel failed", index);
+		 return ret;
+	}
+
+	priv->vrings[index]->enable = 0;
+	return 0;
+}
+
+static int
+virtio_vdpa_virtq_enable(struct virtio_vdpa_priv *priv, int index)
+{
+	int ret;
+	int vid;
+	struct rte_vhost_vring vq;
+	struct virtio_pci_dev_vring_info vring_info;
+	uint64_t gpa;
+
+	vid = priv->vid;
+
+	ret = rte_vhost_get_vhost_vring(vid, index, &vq);
+	if (ret)
+		return ret;
+
+	ret =virtio_pci_dev_interrupts_enable(priv->vpdev, vq.callfd, index+1);
+	if (ret) {
+		 DRV_LOG(ERR, "Virtq interrupt enable failed");
+		 return ret;
+	}
+
+	gpa = virtio_vdpa_hva_to_gpa(vid, (uint64_t)(uintptr_t)vq.desc);
+	if (gpa == 0) {
+		DRV_LOG(ERR, "Virtio vdpa dev %s fail to get GPA for descriptor ring %d.",
+						priv->vdev->device->name, index);
+		return -EINVAL;
+	}
+	DRV_LOG(DEBUG, "Virtq %d desc addr%"PRIx64, index, gpa);
+	priv->vrings[index]->desc = gpa;
+	vring_info.desc = gpa;
+
+	gpa = virtio_vdpa_hva_to_gpa(vid, (uint64_t)(uintptr_t)vq.avail);
+	if (gpa == 0) {
+		DRV_LOG(ERR, "Fail to get GPA for available ring.");
+		return -EINVAL;
+	}
+	DRV_LOG(DEBUG, "Virtq %d avail addr%"PRIx64, index, gpa);
+	priv->vrings[index]->avail = gpa;
+	vring_info.avail = gpa;
+
+	gpa = virtio_vdpa_hva_to_gpa(vid, (uint64_t)(uintptr_t)vq.used);
+	if (gpa == 0) {
+		DRV_LOG(ERR, "Fail to get GPA for used ring.");
+		return -EINVAL;
+	}
+	DRV_LOG(DEBUG, "Virtq %d used addr%"PRIx64, index, gpa);
+	priv->vrings[index]->used = gpa;
+	vring_info.used = gpa;
+
+	//TO_DO: need to check vq_size not exceed hw limit
+	priv->vrings[index]->size = vq.size;
+	vring_info.size = vq.size;
+
+	DRV_LOG(DEBUG, "Virtq %d nr_entrys:%d",index, vq.size);
+	if (virtio_pci_dev_queue_set(priv->vpdev, index, &vring_info)) {
+			DRV_LOG(ERR, "Setup_queue failed");
+			return -EINVAL;
+	}
+
+	ret = virtio_vdpa_virtq_doorbell_relay_enable(priv, index);
+	if (ret) {
+		 DRV_LOG(ERR, "Virtq doorbell relay failed");
+		 return ret;
+	}
+
+	priv->vrings[index]->enable = 1;
+	virtio_pci_dev_queue_notify(priv->vpdev, index);
+	return 0;
+}
+
+static int
+virtio_vdpa_vring_state_set(int vid, int vring, int state)
```

Comment:
please keep the code consistent,

either vring_index or vring or just index

here you say vring and on line 280, you say index.
so please change everything to vq_idx.
it indicates that its an virtiq index.